framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -1410,6 +1549,83 @@ void VulkanReplayConsumerBase::GetMatchingDevice(InstanceInfo* instance_info, Ph
+                                                      std::vector<VkPhysicalDevice>&       replay_device_group)
+{
+    for (auto handle_id : capture_device_group)
+    {
+        auto physical_device_info = GetObjectInfoTable().GetPhysicalDeviceInfo(handle_id);
+        GetMatchingDevice(instance_info, physical_device_info);
+        replay_device_group.emplace_back(physical_device_info->handle);
+    }
+
+    // Modified physical device group to fit requirements(VVL).
+    auto physical_device = physical_device_info->handle;
+    auto result          = std::find(replay_device_group.begin(), replay_device_group.end(), physical_device);
+    bool need_modified   = (result == replay_device_group.end());
+
+    std::vector<VkPhysicalDeviceGroupProperties> replay_group_props;
+
+    if (!need_modified)
+    {
+        VkInstance instance               = instance_info->handle;
+        auto       table                  = GetInstanceTable(instance);
+        uint32_t   replay_gpu_group_count = 0;
+        VkResult   err = table->EnumeratePhysicalDeviceGroups(instance, &replay_gpu_group_count, nullptr);
+        GFXRECON_ASSERT(!err);
+
+        if (replay_gpu_group_count)
+        {
+            replay_group_props.resize(replay_gpu_group_count);
+            err = table->EnumeratePhysicalDeviceGroups(instance, &replay_gpu_group_count, replay_group_props.data());
+            GFXRECON_ASSERT(!err);
+
+            bool find_in_props = false;
+            for (auto& prop : replay_group_props)
+            {
+                bool find_gpu = true;
+                for (auto handle : replay_device_group)
+                {
+                    auto it = std::find(prop.physicalDevices, prop.physicalDevices + prop.physicalDeviceCount, handle);
+                    if (it == (prop.physicalDevices + prop.physicalDeviceCount))
+                    {
+                        find_gpu = false;
+                        break;
+                    }
+                }
+                if (find_gpu)
+                {
+                    find_in_props = true;
+                    break;
+                }
+            }
+            need_modified = !find_in_props;
+        }
+    }
+
+    if (need_modified)
+    {
+        for (auto& prop : replay_group_props)
+        {
+            auto it = std::find(prop.physicalDevices, prop.physicalDevices + prop.physicalDeviceCount, physical_device);
+            if (it != (prop.physicalDevices + prop.physicalDeviceCount))
+            {
+                replay_device_group.clear();
+                std::copy(prop.physicalDevices,
+                          prop.physicalDevices + prop.physicalDeviceCount,
+                          std::back_inserter(replay_device_group));
+                break;
+            }
+        }
+    }
+
+    device_group_create_info.physicalDeviceCount = static_cast<uint32_t>(replay_device_group.size());
+    device_group_create_info.pPhysicalDevices    = replay_device_group.data();
```

Comment:
This assignment means that `device_group_create_info.pPhysicalDevices` can't exist past the destruction of `replay_device_group` since `pPhysicalDevices` would point to freed memory after that.   The chain of ownership from the call chain down to this function doesn't make this relationship clear.  Is there a way to make that more clear, like not pass in `device_group_create_info` and return a bool and let the caller decide what to do with `device_group_create_info.pPhysicalDevices`?

Reply:
I think my suggestion is also awkward, so use your judgment, but I think the lifetime of that pointer needs to be more clear.

Reply:
Added a commit to save `replay_device_group` in `DeviceInfo`.